src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -176,6 +177,53 @@ public int numOfSessionBetween(SessionDate inclusiveStart, SessionDate inclusive
         return (firstInSpan.numOfDayTo(lastInSpan) / interval.getValue()) + 1;
     }
 
+    /**
+     * Checks if the {@code Session} slot overlaps with all recurring sessions.
+     * @param otherSession the other session that is compared to.
+     */
+    public boolean isOverlapping(Session otherSession) {
+        return otherSession.isOverlapping(this);
+    }
+
+    /**
+     * Checks if all {@code RecurringSession} slots overlaps with all recurring sessions.
+     * @param otherSession the other recurring session that is compared to.
+     */
+    public boolean isOverlapping(RecurringSession otherSession) {
+        if (startAfter(otherSession.getSessionDate())) {
+            return otherSession.isOverlapping(this);
+        }
+        int daysBetween = (int) ChronoUnit.DAYS
+                .between(getSessionDate().getDate(), otherSession.getSessionDate().getDate());
+        // Recurrence length in terms of when the first session begins
+        int firstSessionRecurrenceLength =
+                (int) ChronoUnit.DAYS.between(getSessionDate().getDate(), getLastSessionDate().getDate());
+        int secondSessionRecurrenceLength =
+                (int) ChronoUnit.DAYS.between(otherSession.getSessionDate().getDate(),
+                        otherSession.getLastSessionDate().getDate()) + daysBetween;
+        // Days from when the earlier session starts
+        int daysFromFirstSessionStart = getInterval().getValue();
+        int daysFromSecondSessionStart = daysBetween;
+        while (daysFromFirstSessionStart <= firstSessionRecurrenceLength
+                || daysFromSecondSessionStart <= secondSessionRecurrenceLength) {
```

Comment:
BUG! must be && we keep doing while both conditions are true.
if any one condition false, we get out of while loop

Reply:
Nonsense i said